### PR TITLE
[wip] Add option to enforce membership of a project

### PR DESCRIPTION
**Pull request by kassybas**
Previously when removing a group from the `group:` part of the configuration, it was not removed from GitLab.

With `enforce_group_shares` option set to true, you can ensure that only the defined groups can have access to the projects.

### CODE REVIEW

**Comment by gdubicki** [comment]
Hey @kassybas ! Thanks for working on this contribution. Please ping me if I can help you move this forward in any way!

**Comment by gdubicki** [comment]
Hey @kassybas ! Please see https://github.com/egnyte/gitlabform/pull/83 as it relates to what you are working on here. Note that you will most probably need to rebase because of that, sorry!

File: version [hunk 1]
```diff
@@ -1 +1 @@
-1.9.1
\ No newline at end of file
+1.9.2
```
Comment (gdubicki) [inline version]: Please do not uptick the version in PRs. See https://github.com/egnyte/gitlabform/commit/87819c313f36748d988fe3a08b862da4112a4661 for a rationale.

**Comment by gdubicki** [comment]
Hi @kassybas ! Because of #68 we badly needed better management of group membership, because it's used as a workaround for the GitLab bug (see https://github.com/egnyte/gitlabform/issues/68#issuecomment-581003703). Therefore I have just implemented something similar to what you did here and am releasing it in v1.13.0.

However you implementation looks nicer so it would be cool if you would complete this MR and get this merged after rebasing!

Please ping me if you need any help with that.

**Comment by kassybas** [comment]
Hi @gdubicki,
I started working on rebasing to your changes, but since then, GitLab made it possible to share groups with other groups, so it is not required for us to use Gitlabform. 

Using the Gitlab Terraform provider can handle the necessary configurations, and Terraform fits our current toolchain better.

**Pull request closed without merging**